frappe/integrations/doctype/social_login_key/social_login_key.py

```diff
@@ -80,6 +80,8 @@ def validate(self):
 			frappe.throw(
 				_("Please enter Client Secret before social login is enabled"), exc=ClientSecretNotSetError
 			)
+		if self.social_login_provider == "Keycloak":
+			self.api_endpoint = self.base_url+"/protocol/openid-connect/userinfo"
```

Comment:
Any reason to set this here, could this be done in `get_social_login_provider`?

Reply:
The reason for implementing the logic in the validate function is to ensure that the API endpoint dynamically adjusts based on changes to the base URL. When the base URL is modified, it's imperative that the corresponding API endpoint updates accordingly. However, during the auto fill process, users can only edit the base URL and are unable to directly modify the API endpoint.

To ensure that the API endpoint reflects the updated base URL, the logic must be implemented in the validate function, as it is the only function executed after the save button is clicked. This approach guarantees that any changes to the base URL are correctly propagated to the API endpoint, ensuring consistency. Implementing this logic in the validate function is more effective than placing it in the get_social_login_key function.